pkg/receiver/gitproviderreceiver/internal/scraper/gitlabscraper/gitlab_scraper.go

```diff
@@ -139,32 +177,57 @@ func (gls *gitlabScraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
 		}
 	}
 
-	ch := make(chan projectData)
+	var wg1 sync.WaitGroup
+	var wg2 sync.WaitGroup
 
-	var wg sync.WaitGroup
+	branchCh := make(chan projectData)
 
 	// TODO: Must account for when there are more than 100,000 branch names in a project.
 	for _, project := range projectList {
-		wg.Add(1)
-
-		// created shadowed declaration due to loop variable because Loop variables
-		// captured by 'func' literals in 'go' statements might have unexpected values
-		project := project
-		go func() {
-			gls.getBranches(ctx, graphClient, project.Path, ch, &wg)
-		}()
+		wg1.Add(1)
+
+		go func(project gitlabProject) {
+			gls.getBranches(ctx, graphClient, project.Path, branchCh, &wg1)
+		}(project)
+	}
+
+	go func() {
```

Comment:
In these go routines, we should add buffered channels to reduce the need for extreme resources caused by multiple threads.